Adding ToShortDateString() and sisters to Date and TimeOfDay.

rafidka opened the pull request:
Added the following convenience methods to Date and TimeOfDay:
- Date.ToShortDateString()
- Date.ToLongDateString()
- TimeOfDay.ToShortTimeString()
- TimeOfDay.ToLongDateString()

#16

rafidka commented:
@mj1856, can you please review the changes I did above and let me know if you are OK with me merging the pull request. Then after that I will update the tests following your change of project to non-PCL.

mj1856 commented:
Can you please rebase to remove the merge commit?

```
git fetch upstream
git checkout master
git rebase upstream/master
git push --force
```

Also, in the future, you might want to consider creating a feature branch in your fork for different items, that makes it much cleaner to do separate PRs for different issues simultaneously.

rafidka commented:
Thanks @mj1856. I changed ToIsoString() to use invariant culture and I rebased the repository as you suggested. I think we are good to merge this PR now, but I will leave this for you.
